@types/swagger-client/index.d.ts

```diff
@@ -37,13 +37,26 @@ declare module 'swagger-client' {
     responses: { [responseStatus: string]: Json };
   }
 
-  export interface Parameter {
+  interface ParameterBase {
     name: string;
+    in: 'path' | 'query' | 'header' | 'cookie';
+    description?: string;
+    required?: boolean;
     example: Json;
-    required: boolean;
+  }
+
+  export interface Parameter extends ParameterBase {
     schema: SchemaObject;
   }
 
+  export interface Parameter extends ParameterBase {
+    content: {
+      [name: string]: {
+        schema: SchemaObject;
+      };
+    };
+  }
```

Comment:
Typescript ends up merging these types, so this is functionally the same as writing
```ts
export interface Parameter extends ParameterBase {
    schema: SchemaObject;
    content: {
      [name: string]: {
        schema: SchemaObject;
      };
    };
  }
```
This doesn't cause problems for us the way the code is written (for some reason, not sure why ts doesn't complain about unreachable code) though. It comes back to how correct we want these types to be